src/util/gentle-register-property.js

```diff
@@ -49,41 +70,34 @@ export default function gentleRegisterProperty (property, meta = {}, window = gl
 		definition.initialValue = INITIAL_VALUES[definition.syntax];
 	}
 
-	try {
-		window.CSS.registerProperty(definition);
+	_properties.add(property);
+
+	_layerRules += `@property ${property} {
+		syntax: "${definition.syntax}";
+		inherits: ${definition.inherits};
+		${definition.initialValue !== undefined ? `initial-value: ${definition.initialValue};` : ""}
+	}\n`;
+
+	if (!_styleSheet && window.document.adoptedStyleSheets && !Object.isFrozen(window.document.adoptedStyleSheets)) {
+		// The modern adoptedStyleSheets API is supported
+		_styleSheet = new CSSStyleSheet();
+		window.document.adoptedStyleSheets.push(_styleSheet);
 	}
-	catch (e) {
-		let error = e;
-		let rethrow = true;
 
-		if (e instanceof window.DOMException) {
-			if (e.name === "InvalidModificationError") {
-				// Property is already registered, which is fine
-				rethrow = false;
-			}
-			else if (e.name === "SyntaxError") {
-				// In Safari < 18.2 (where we face the infinite loop bug),
-				// there is no way to provide an initial value for a custom property with a syntax of "<string>".
-				// There will always be an error: “The given initial value does not parse for the given syntax.”
-				// So we try again with universal syntax.
-				// We do the same for any other syntax that is not supported.
-				definition.syntax = "*";
+	if (!_styleSheet && !_styleElement) {
+		// The modern adoptedStyleSheets API is not supported
+		_styleElement = Object.assign(window.document.createElement("style"), { id: "style-observer-styles" });
+		window.document.head.append(_styleElement);
+	}
 
-				try {
-					window.CSS.registerProperty(definition);
-					rethrow = false;
-				}
-				catch (e) {
-					error = e;
-				}
-			}
-		}
+	let rule = `@layer style-observer-registered-properties {
+		${_layerRules}
```

Comment:
I might be misunderstanding this code, but in case: you can have multiple layers with the same name and they're all part of the same layer. No need to rewrite the layer each time.

Reply:
You are absolutely right: I didn't think of layers with the same name. That's why I didn't use `adoptCSS()` here in the first place. I'll refactor it.